CMakeLists.txt

```diff
@@ -15,15 +15,19 @@ include_directories(${BOOST_INCLUDE_DIR})
 pkg_check_modules(EIGEN3 REQUIRED "eigen3")
 include_directories(${EIGEN3_INCLUDE_DIRS})
 
-find_package(SISL)
-if (SISL_FOUND)
-    pkg_check_modules(BASELIB REQUIRED "base-lib")
-    include_directories(${BASELIB_INCLUDE_DIRS})
+pkg_check_modules(BASELIBSISL "base-lib-sisl")
+if(BASELIBSISL_FOUND)
+    find_package(SISL REQUIRED)
     include_directories(${SISL_INCLUDE_DIRS})
     list(APPEND TOOLKIT_ADDITIONAL_SOURCES ${CMAKE_SOURCE_DIR}/src/spline.cpp)
-    list(APPEND TOOLKIT_ADDITIONAL_LIBRARIES ${BASELIB_LIBRARIES} ${SISL_LIBRARIES})
+    list(APPEND TOOLKIT_ADDITIONAL_LIBRARIES ${SISL_LIBRARIES})
+    list(APPEND PKG_CFLAGS "-DWITH_SISL=1")
 endif()
 
+pkg_check_modules(BASELIB REQUIRED "base-lib")
+include_directories(${BASELIB_INCLUDE_DIRS})
+list(APPEND TOOLKIT_ADDITIONAL_LIBRARIES ${BASELIB_LIBRARIES})
```

Comment:
This toolkit_additional_libraries is puzzling me ...

Reply:
This has been part of the previously sisl-dependent block. Since it is not optional anymore I updated it. So 'puzzling' means 'is not needed'?

Reply:
I got that it was there ... but I wonder why (if ?) it is needed.

Reply:
All types are contained in base-types library. So I removed the base-lib part.